application/client.js

```diff
@@ -152,7 +152,14 @@ module.exports = HttpGateway.extend({
                     try {
                         responseData = JSON.parse(responseText);
                     } catch (e) {
-                        responseData = responseText;
```

Comment:
Yes, but (IIRC) when I stepped through the response, the actual response text and code were discarded in that library and a useless response object was given back (with 0 code).  It really should have thrown an exception/error instead IMO.